packages/erk-shared/tests/unit/integrations/pr/test_submit.py

```diff
@@ -0,0 +1,276 @@
+"""Unit tests for core PR submission operation.
+
+Tests the execute_core_submit() function which handles git push + gh pr create
+without requiring Graphite.
+"""
+
+from dataclasses import dataclass
+from pathlib import Path
+
+from erk_shared.git.abc import Git
+from erk_shared.git.fake import FakeGit
+from erk_shared.github.abc import GitHub
+from erk_shared.github.fake import FakeGitHub
+from erk_shared.github.types import PRDetails, PullRequestInfo
+from erk_shared.integrations.graphite.abc import Graphite
+from erk_shared.integrations.graphite.fake import FakeGraphite
+from erk_shared.integrations.gt.events import CompletionEvent, ProgressEvent
+from erk_shared.integrations.pr.submit import execute_core_submit
+from erk_shared.integrations.pr.types import CoreSubmitError, CoreSubmitResult
+
+
+@dataclass
+class FakePrKit:
+    """Fake PrKit implementation for testing.
+
+    This satisfies the PrKit Protocol using concrete Fake implementations.
+    """
+
+    git: Git
+    github: GitHub
+    graphite: Graphite
+
```

Comment:
Yeah I don't see why we need this. We should just make ErkContext easier to use in tests (if that is the problem)